common/git.go

```diff
@@ -59,28 +57,133 @@ func (r *realGitOps) gitRootPath(fromPath string) (string, error) {
 	cmd.Dir = fromPath
 	output, err := cmd.Output()
 	if err != nil {
-		return "", err
+		log.Println("Unable to determine Git root path")
+		log.Printf("Checking if the fromPath: %s directory is the root\n", fromPath)
+		// if fromPath contains .git, then it is the root
+		// otherwise, return the error
+		if _, errNotExist := os.Stat(fromPath + "/.git"); os.IsNotExist(errNotExist) {
+			// check if the current working directory is the root
+			cwd, errGetCwd := os.Getwd()
+			if errGetCwd != nil {
+				return "", fmt.Errorf("failed to determine the Git root path: %s %v", output, err)
+			} else {
+				log.Println("fromPath is not the root")
```

Comment:
you probably meant to print the value of `fromPath` and not the actual string?
But maybe we don't need the log print at all anymore?

Reply:
The log message that precedes this is `log.Printf("Checking if the fromPath: %s directory is the root\n", fromPath)` so didn't bother printing again